drivers/dai/intel/ssp/ssp.c

```diff
@@ -1913,47 +1915,46 @@ static int dai_ssp_remove(struct dai_intel_ssp *dp)
 	return 0;
 }
 
-static int dai_ssp_probe_wrapper(const struct device *dev)
+static int ssp_pm_action(const struct device *dev, enum pm_device_action action)
 {
 	struct dai_intel_ssp *dp = (struct dai_intel_ssp *)dev->data;
-	k_spinlock_key_t key;
-	int ret = 0;
-
-	key = k_spin_lock(&dp->lock);
-
-	if (dp->sref == 0) {
-		ret = dai_ssp_probe(dp);
+	switch (action) {
+	case PM_DEVICE_ACTION_SUSPEND:
+		dai_ssp_remove(dp);
+		break;
+	case PM_DEVICE_ACTION_RESUME:
+		dai_ssp_probe(dp);
+		break;
+	default:
+		return -ENOTSUP;
 	}
 
-	if (!ret) {
-		dp->sref++;
-	}
+	return 0;
+}
 
-	k_spin_unlock(&dp->lock, key);
+static int dai_ssp_probe_wrapper(const struct device *dev)
+{
+	int ret;
 
+	ret = pm_device_runtime_get(dev);
 	return ret;
 }
 
 static int dai_ssp_remove_wrapper(const struct device *dev)
 {
-	struct dai_intel_ssp *dp = (struct dai_intel_ssp *)dev->data;
-	k_spinlock_key_t key;
-	int ret = 0;
-
-	key = k_spin_lock(&dp->lock);
-
-	if (--dp->sref == 0) {
-		ret = dai_ssp_remove(dp);
-	}
-
-	k_spin_unlock(&dp->lock, key);
+	int ret;
 
+	ret = pm_device_runtime_put(dev);
 	return ret;
 }
```

Comment:
Can'you simplify the code removing this function and calling `pm_device_reuntime_put()` directly ? This function is not doing anything else now.

Reply:
ack, done